test/e2e/specs/editor/blocks/table.spec.js

```diff
@@ -0,0 +1,274 @@
+		await expect( footerSwitch ).toBeHidden();
+
+		// // Create the table.
+		await page.click( 'role=button[name="Create Table"i]' );
+
+		// Expect the header and footer switches to be present now that the table has been created.
+		await expect( headerSwitch ).toBeVisible();
+		await expect( footerSwitch ).toBeVisible();
+
+		// // Toggle on the switches and add some content.
+		await headerSwitch.check();
+		await footerSwitch.check();
+
+		await page.click(
+			'role=rowgroup >> nth=0 >> role=textbox[name="Header cell text"i] >> nth=0'
+		);
+		await page.keyboard.type( 'header' );
+
+		await page.click(
+			'role=rowgroup >> nth=1 >> role=textbox[name="Body cell text"i] >> nth=0'
+		);
+		await page.keyboard.type( 'body' );
+
+		await page.click(
+			'role=rowgroup >> nth=2 >> role=textbox[name="Footer cell text"i] >> nth=0'
+		);
+		await page.keyboard.type( 'footer' );
+
+		// Expect the table to have a header, body and footer with written content.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+
+		// Toggle off the switches.
+		await headerSwitch.uncheck();
+		await footerSwitch.uncheck();
+
+		// Expect the table to have only a body with written content.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'allows adding and deleting columns across the table header, body and footer', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.insertBlock( { name: 'core/table' } );
+		await editor.openDocumentSettingsSidebar();
+
+		// Create the table.
+		await page.click( 'role=button[name="Create Table"i]' );
+
+		// Toggle on the switches and add some content.
+		await page.locator( 'role=checkbox[name="Header section"i]' ).check();
+		await page.locator( 'role=checkbox[name="Footer section"i]' ).check();
+		await page.click( 'role=textbox[name="Body cell text"i] >> nth=0' );
+
+		// Add a column.
+		await editor.clickBlockToolbarButton( 'Edit table' );
+		await page.click( 'role=menuitem[name="Insert column after"i]' );
+
+		// Expect the table to have 3 columns across the header, body and footer.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+
+		await page.click( 'role=textbox[name="Body cell text"i] >> nth=0' );
+
+		// Delete a column.
+		await editor.clickBlockToolbarButton( 'Edit table' );
+		await page.click( 'role=menuitem[name="Delete column"i]' );
+
+		// Expect the table to have 2 columns across the header, body and footer.
+		expect( await editor.getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	test( 'allows columns to be aligned', async ( { editor, page } ) => {
+		await editor.insertBlock( { name: 'core/table' } );
+
+		await page.click( 'role=spinbutton[name="Column count"i]' );
+		await page.keyboard.press( 'Backspace' );
+		await page.keyboard.type( '4' );
+
+		// Create the table.
+		await page.click( 'role=button[name="Create Table"i]' );
+
+		// Click the first cell and add some text. Don't align.
+		const cells = await page.locator(
```

Comment:
Locator doesn't return a promise.

```suggestion
		const cells = page.locator(
```